xapian-core/weight/pl2plusweight.cc

```diff
@@ -70,23 +72,38 @@ PL2PlusWeight::init(double)
     cl = param_c * get_average_length();
 
     double base_change(1.0 / log(2.0));
-    double mean = double(get_collection_freq()) / get_collection_size();
+    mean = double(get_collection_freq()) / get_collection_size();
     P1 = mean * base_change + 0.5 * log2(2.0 * M_PI);
     P2 = log2(mean) + base_change;
 
     double wdfn_lower = log2(1 + cl / get_doclength_upper_bound());
     double wdfn_upper =
 	get_wdf_upper_bound() * log2(1 + cl / get_doclength_lower_bound());
 
+    double P_delta = P1 + (param_delta + 0.5) * log2(param_delta) - P2 * param_delta;
+    dw = P_delta / (param_delta + 1.0);
+
     // Calculate the lower bound on the weight.
     double P_min =
 	P1 + (wdfn_lower + 0.5) * log2(wdfn_lower) - P2 * wdfn_lower;
-    lower_bound = get_wqf() * P_min / (wdfn_upper + 1.0);
+    // Calculate P_delta_min by setting param_delta to the
+    // lowest permitted value of 0.1.
+    double P_delta_min = P1 + (0.1 + 0.5) * log2(0.1) - P2 * 0.1;
+    // Calculate dw_min by setting param_delta to 1.5 based
+    // on the prescribed the value range for it in weight.h
+    double dw_min = P_delta_min / (1.5 + 1.0);
```

Comment:
So we don't want `P_delta_min`, `P_delta_max` (and similarly `dw_min` and `dw_max`). Is that right?